### DISCUSSION TITLE
Service Apache can not start. Reason:

### DISCUSSION THREAD

**Discussion by reiniermybeats** [leokhoa/laragon#760] [Q&A]
![image](https://github.com/leokhoa/laragon/assets/115628319/d7642420-0522-486f-8f4d-dc16889e7e5f)

it says i am missing some files, but when i download them and try to install them the installer from microsoft says it is already installed.

this error came from nowhere. i did not change any software or settings. i just switched php versions and this error popped up. now all 3 php versions i have installed dont work anymore. so my development environment is sadly unusable.

update:
a fresh install works. but when redownloading my php versions and mariadb from php.net and mariadb.com and using them in the fresh install i get the error again.

**Comment by halimkun** [comment]
From the message, you may need to install `Microsot Visual C++ 2015` which might solve your problem.
download [MS Visual C++ 2015](https://www.microsoft.com/en-us/download/details.aspx?id=48145)

**Reply by reiniermybeats** [reply]
thanks, but from my explanation below the image:

> when i download them (redistributables) and try to install them the installer from **microsoft says it is already installed**.

i have since downloaded a fresh install of laragon in a separate directory. fresh php. fresh mariadb. etc. of all my required versions.

it seems to work again. but a pain to do.
and a very strange error... 
is this just windows being windows?

**Comment by popkilli** [comment]
**Microsoft Visual C++ 2015-2022 Redistributable Package 14.38.33135.0**

https://aka.ms/vs/17/release/vc_redist.x86.exe
https://aka.ms/vs/17/release/vc_redist.x64.exe

_Reference: https://learn.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist?view=msvc-170#visual-studio-2015-2017-2019-and-2022_

**Reply by reiniermybeats** [reply]
thanks that one does let me install it. just installed it and after restarting pc and laragon i get the same error
i think it might be a setting or file i messed up in that install. so i wont be trying to debug this anymore.
my new fresh laragon install works fine.

**Comment by reiniermybeats** [comment]
no longer going to debug this. fresh install fixed my problem

**Comment by SebasBaezCode** [comment]
To resolve this issue, in my particular case, it was about ensuring that the versions of Apache, PHP, and their modules are all compatible with each other. If you change PHP or Apache versions that use different modules, Laragon may falsely alert that a certain version is not installed, when in fact it is installed, but there might be internal mismatches.

For example, I had Apache -VS16 installed, so I switched it to Apache -VC14, which matches the version installed on my machine. I also checked the PHP versions.

I hope this helps someone.

![image](https://github.com/user-attachments/assets/7ebb7ea9-a5b2-44f7-8156-204611c6a342)

![image](https://github.com/user-attachments/assets/817fdc31-26c1-4564-9bfb-99168e396984)
